### PR TITLE
fix: validation errors in signup form

### LINKED ISSUE #1172: Bug: Validation error in signup form
### Description

Validation error should come inside form only rather than in form Toast massage , it will help user to catch error easily.

### Proposed Solution

rather than using toast make object for every feild-state that can contains validation error as well so we can show error just below of the fields.

### Screenshots

_No response_

### Browser 🥦

Google Chrome

### Checklist ✅

- [X] I checked and didn't find similar [issue](https://github.com/MilanCommunity/Milan/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/MilanCommunity/Milan/blob/main/CONTRIBUTING.md)
- [ ] I am willing to work on this issue (blank for no).

### PR DESCRIPTION
**Pull request by Endless8**
closes #1172

### 👷🏻 Changes made  
I removed the toast notifications, assigned the validation errors to formState and output them below the corresponding field adding a field property to all the errors for filtering.

### 📸 Screenshots 
image

## Summary by CodeRabbit

- **Refactor**
  - Streamlined form error handling by setting form errors directly in the state instead of individual toast notifications.

- **New Features**
  - Introduced specific field error messages in the signup form for improved user guidance.

- **Style**
  - Added new styles for displaying error messages on the authentication page for better visibility.

- **Bug Fixes**
  - Enhanced website field validation to ensure the input is a valid URL.

### CODE REVIEW

File: src/hooks/useValidation.js [hunk 1]
```diff
@@ -97,75 +144,113 @@ const useValidation = (credentials, userSignup, clubSignup) => {
...
       errors.push({
         error: true,
         message: "Please enter your club iframe code",
+        field: "iframe",
       });
     }
   }
 
   if (credentials.website) {
     if (!urlRegex.test(credentials.website)) {
```
Comment (tamalCodes) [inline src/hooks/useValidation.js:161]:
```suggestion
  if (credentials.website && !urlRegex.test(credentials.website)) {
```

**Review by tamalCodes — CHANGES_REQUESTED** [review]

File: src/hooks/useValidation.js (hunk 1 above)
Comment (tamalCodes) [reply src/hooks/useValidation.js:161]: Also please fix this issue (Brave browser)
image

Comment (Endless8) [reply src/hooks/useValidation.js:161]: Do you mean the overlapping of text and eye icon?

Comment (tamalCodes) [reply src/hooks/useValidation.js:161]: As you can see, the eye icon is not in place if the errors appear. The eye icon should be inside the box.

Comment (Endless8) [reply src/hooks/useValidation.js]: The suggested edits are done, thanks

**Pull request merged**